src/system/NewForm/FormAutocomplete.js

```diff
@@ -153,29 +160,44 @@ const FormAutocomplete = React.forwardRef(
 			[ getAllOptions, optionLabel ]
 		);
 
+		// this method gets called when we confirm the selection via click/enter
 		const onValueChange = useCallback(
 			inputValue => {
 				if ( inputValue ) {
+					setSelectedValue( inputValue );
+					setInputQuery( inputValue );
+					onChange( getOptionByLabel( inputValue ), inputValue );
+					setIsDirty( false );
+				} else if ( resetOnNoMatch && inputQuery !== selectedValue ) {
+					// reset the content if there's no match
+					setSelectedValue( '' );
+					setInputQuery( '' );
```

Comment:
> The only comment I have is if we should set the value as '' or null, because the options are objects, not simple strings.
@djalmaaraujo you mean here? Keep in mind that selectedValue/InputQuery are string or simple types. The object is the one we pass on the `onChange`. Which is retrieved through the `getOptionByLabel`

Reply:
@djalmaaraujo with the latest changes I'm sending `null` as you proposed.